packages/cli/test/unit/commands/addons/wait.unit.test.ts

```diff
@@ -0,0 +1,338 @@
+          .reply(404, {id: 'not_found', message: 'Not found.'})
+        await runCommand(Cmd, [
+          '--wait-interval',
+          '1',
+          'www-redis-2',
+        ])
+        expectOutput(stderr.output, `
+Destroying www-redis-2...
+Destroying www-redis-2... done
+`)
+        expectOutput(stdout.output, '')
+      })
+      it('does NOT notify the user when deprovisioning takes less than 5 seconds', async () => {
+        const notifySpy = sandbox.spy(require('@heroku-cli/notifications'), 'notify')
+        const deprovisioningAddon = _.clone(fixtures.addons['www-redis-2'])
+        deprovisioningAddon.id = '37f27548-db4a-4ae0-bb48-57125df0ddc2'
+        deprovisioningAddon.name = 'www-redis-3'
+        nock('https://api.heroku.com')
+          .post('/actions/addons/resolve', {app: null, addon: 'www-redis-3'})
+          .reply(200, [deprovisioningAddon])
+          .get('/apps/acme-inc-www/addons/www-redis-3')
+          .reply(404, {id: 'not_found', message: 'Not found.'})
+        await runCommand(Cmd, [
+          '--wait-interval',
+          '1',
+          'www-redis-3',
+        ])
+        expect(notifySpy.called).to.be.false
+        expect(notifySpy.calledOnce).to.be.false
+      })
+      it('notifies the user when provisioning takes longer than 5 seconds', async () => {
+        const notifySpy = sandbox.spy(require('@heroku-cli/notifications'), 'notify')
+        const deprovisioningAddon = _.clone(fixtures.addons['www-redis-2'])
+        deprovisioningAddon.id = '967dff74-99b4-4fd2-a0f0-79b523d5c0e1'
+        deprovisioningAddon.name = 'www-redis-4'
+        nock('https://api.heroku.com')
+          .post('/actions/addons/resolve', {app: null, addon: 'www-redis-4'})
+          .reply(200, [deprovisioningAddon])
+          .get('/apps/acme-inc-www/addons/www-redis-4')
+          .reply(200, () => {
+            clock.tick(5000)
+            return deprovisioningAddon
+          })
+          .get('/apps/acme-inc-www/addons/www-redis-4')
+          .reply(404, {id: 'not_found', message: 'Not found.'})
+        await runCommand(Cmd, [
+          '--wait-interval',
+          '1',
+          'www-redis-4',
+        ])
+        expect(notifySpy.called).to.be.true
+        expect(notifySpy.calledOnce).to.be.true
+      })
+    })
+  })
+  context('waiting for add-ons', () => {
+    context('for an app', () => {
+      it('waits for addons serially', async () => {
+        const ignoredAddon = _.clone(fixtures.addons['www-db'])
+        ignoredAddon.state = 'provisioned'
+        const wwwAddon = _.clone(fixtures.addons['www-db'])
+        wwwAddon.state = 'provisioning'
+        const redisAddon = _.clone(fixtures.addons['www-redis'])
+        redisAddon.state = 'provisioning'
+        const redis2Addon = _.clone(fixtures.addons['www-redis-2'])
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        redis2Addon.state = 'deprovisioning'
```

Comment:
I'm ok with this. The only way to avoid this will be to update `@heroku-cli/schema`, which is out of scope for this work.

Reply:
yep, makes sense.